indexer-opts/src/lib.rs

```diff
@@ -0,0 +1,298 @@
+use std::str::FromStr;
+
+use bigdecimal::{FromPrimitive, ToPrimitive};
+pub use clap::{self, Parser};
+use tracing_subscriber::EnvFilter;
+
+use near_jsonrpc_client::{methods, JsonRpcClient};
+use near_lake_framework::near_indexer_primitives::types::{BlockReference, Finality};
+
+pub const LOGGING_PREFIX: &str = "indexer_opts";
+
+/// NEAR Indexer Opts
+/// Start options for NEAR micro indexers
+#[derive(Parser, Debug)]
+#[clap(
+    version,
+    author,
+    about,
+    disable_help_subcommand(true),
+    propagate_version(true),
+    next_line_help(true)
+)]
+pub struct Opts {
+    /// Enabled Indexer for Explorer debug level of logs
+    #[clap(long)]
+    pub debug: bool,
+    /// Sets the micro-indexer instance ID (for reading/writing indexer meta-data)
+    #[clap(long, env)]
+    pub indexer_id: String,
+    /// Block height to start the stream from
+    #[clap(long, short, env)]
+    pub start_block_height: u64,
```

Comment:
Currently, if we use `--start-mode from-latest` this parameter is not used. Maybe we might use this param as a fallback. At the moment we would panic if we couldn't get the final block from RPC.